app/main.py

```diff
@@ -1,3 +1,57 @@
+from __future__ import annotations
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    @staticmethod
+    def is_distance(some_data: Distance | int | float) -> Any:
+        if isinstance(some_data, Distance):
+            return some_data.km
+        else:
+            return some_data
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        return Distance(self.km + self.is_distance(other))
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        self.km += self.is_distance(other)
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        if not isinstance(other, (int, float)):
+            raise TypeError(
+                f"Unsupported operation class Distance * {type(other)}"
+            )
```

Comment:
If you already added a check, why we are checking only at `__mul__` and `__truediv__` methods?

Reply:
Staticmethod above is not bool (returns data). May be it's possible to use it in __mul__ and __truediv__ methods, but code became knotty (cause only in this methods we have to raise error message). I recompose methods to avoid using 'not', but prefer to keep this construction without using staticmethod here. Are you agree?

Reply:
In this situation no need to remove it because it's logical. Thanks for explaining your solution here, and yep, I agree with you